flexbox/src/main/java/com/google/android/flexbox/FlexItem.java

```diff
@@ -0,0 +1,241 @@
+     *
+     * @param height the height attribute. Can be one of the
+     * constants MATCH_PARENT(-1) or WRAP_CONTENT(-2), or an exact size.
+     */
+    void setHeight(int height);
+
+    /**
+     * @return the order attribute of the flex item.
+     *
+     * The attribute can change the ordering of the children views are laid out.
+     * By default, children are displayed and laid out in the same order as they appear in the
+     * layout XML. If not specified, {@link #ORDER_DEFAULT} is set as a default value.
+     */
+    int getOrder();
+
+    /**
+     * Sets the order attribute to the flex item
+     *
+     * @param order the order attribute
+     */
+    void setOrder(int order);
+
+    /**
+     * @return the flex grow attribute of the flex item
+     *
+     * The attribute determines how much this child will grow if positive free space is
+     * distributed relative to the rest of other flex items included in the same flex line.
+     * If not specified, {@link #FLEX_GROW_DEFAULT} is set as a default value.
+     */
+    float getFlexGrow();
+
+    /**
+     * Sets the flex grow attribute to the flex item
+     *
+     * @param flexGrow the flex grow attribute
+     */
+    void setFlexGrow(float flexGrow);
+
+    /**
+     * @return the flex shrink attribute of the flex item
+     *
+     * The attributes determines how much this child will shrink is negative free space is
+     * distributed relative to the rest of other flex items included in the same flex line.
+     * If not specified, {@link #FLEX_SHRINK_DEFAULT} is set as a default value.
+     */
+    float getFlexShrink();
+
+    /**
+     * Sets the flex shrink attribute to the flex item
+     *
+     * @param flexShrink the flex shrink attribute
+     */
+    void setFlexShrink(float flexShrink);
+
+    /**
+     * @return the align self attribute of the flex item
+     *
+     * The attribute determines the alignment along the cross axis (perpendicular to the
+     * main axis). The alignment in the same direction can be determined by the
+     * align items attribute in the parent, but if this is set to other than
+     * {@link AlignSelf#AUTO}, the cross axis alignment is overridden for this child.
+     * The value needs to be one of the values in ({@link AlignSelf#AUTO},
+     * {@link AlignItems#STRETCH}, {@link AlignItems#FLEX_START}, {@link
+     * AlignItems#FLEX_END}, {@link AlignItems#CENTER}, or {@link AlignItems#BASELINE}).
+     * If not specified, {@link AlignSelf#AUTO} is set as a default value.
+     */
+    @AlignSelf
+    int getAlignSelf();
+
+    /**
+     * Sets the align self attribute to the flex item
+     *
+     * @param alignSelf the order attribute
+     */
+    void setAlignSelf(@AlignSelf int alignSelf);
+
+    /**
+     * @return the minimum width attribute of the flex item
+     *
+     * The attribute determines the minimum width the child can shrink to.
+     */
+    int getMinWidth();
```

Comment:
I see. I think some custom Views have their own minimum width/height specified, and FlexboxLayout might want to be mindful of them. It is up to you, though.
+2 otherwise.